hpx/async.hpp

```diff
@@ -101,12 +101,12 @@ namespace hpx { namespace detail
         template <typename F, typename ...Ts>
         BOOST_FORCEINLINE static
         typename boost::enable_if_c<
-            traits::detail::is_deferred_callable<F(Ts...)>::value,
-            hpx::future<typename util::deferred_call_result_of<F(Ts...)>::type>
+            traits::detail::is_deferred_callable<F(Ts&&...)>::value,
+            hpx::future<typename util::detail::deferred_result_of<F(Ts&&...)>::type>
         >::type
         call(BOOST_SCOPED_ENUM(launch) launch_policy, F&& f, Ts&&... ts)
         {
-            typedef typename util::deferred_call_result_of<
+            typedef typename util::detail::deferred_result_of<
                 F(Ts...)
```

Comment:
I am a little confused. Could you please sched some light on where we need to add ``&&`` and where not?
In particular, you have, for example, ```typename util::detail::deferred_result_of<F(Ts&&...)>::type``` in line 105 and ```typename util::detail::deferred_result_of<F(Ts...)>::type``` right here (line 109). Could you pleas elaborate on that?